marklogic-data-hub-central/ui/src/pages/Browse.tsx

```diff
@@ -148,13 +148,19 @@ const Browse: React.FC<Props> = ({location}) => {
     if (searchOptions.nextEntityType && searchOptions.nextEntityType !== "All Data") {
       setCardView(false);
     }
-    if ((entities.length || (searchOptions.nextEntityType === "All Data" || searchOptions.nextEntityType === "All Entities" || searchOptions.nextEntityType === undefined)) &&
-      (!searchOptions.nextEntityType ||
-        (searchOptions.nextEntityType === "All Entities" && !searchOptions.entityTypeIds.length && !searchOptions.selectedTableProperties.length && !cardView) ||
+    if ((entities.length > 0 || (searchOptions.nextEntityType === "All Data" || searchOptions.nextEntityType === "All Entities" || searchOptions.nextEntityType === undefined)) &&
```

Comment:
Something to consider here - there are a lot of conditionals here, making it hard for a human to read this and understand what the conditions are. 

The Clean Code book recommends defining local variables to capture the names of the conditions here, which then makes the overall conditional much easier to read. I think that would help a lot here. 

For example, the first condition above is saying - if at least one entity type exists (I am assuming that "entities" is really "entityTypes" or the user has selected All Data or All Entities or has not selected anything (and I'm assuming that "nextEntityType" is really "what the user selected" - "next" is confusing here). So how about:

    const entityTypesExistOrNoEntityTypeIsSelected = (all the above logic)

Reading that... I'm wondering when that would ever be false?? Maybe "nextEntityType" isn't really "what the user has selected in the search type dropdown"? I'm confused.

Regardless, It looks like all of the logic can then be simplified to something that reads like this:

    if (entityTypesExistOrNoEntityTypeIsSelected && (nameForFirstCondition || nameForSecondCondition || nameForThirdCondition || nameForFourthCondition))

Which would be far easier to understand than the current code.

Reply:
Refactored the code a bit to better understand the logic.